Enable Console lib split

markstory opened the pull request:
Key changes here:

* The `ConsoleErrorHandler` has moved to the Error package. This lets us have much simpler dependencies in the Console package.
* The Command base class includes a bunch of integrations with the rest of CakePHP that make it burdensome to use. By adding a CommandInterface we can narrow the dependencies for standalone usage.

Applications that use the console standalone library would need to setup their own `run()` method that uses the optionparser if they want. I'm open to adding a leaner Command base class but don't have a good name for it right now. Any suggestions are welcome.

markstory committed:
Move ConsoleErrorHandler to Error

Having this class in the Error package makes the splits for Console
libraries much simpler as we don't need a split for Routing and Error.

markstory committed:
Add missing class alias.

markstory committed:
Add CommandInterface and reference it.

The Command base class includes a bunch of integrations with the rest of
CakePHP that make it burdensome to use. By adding a CommandInterface we
can narrow the dependencies for standalone usage.

Applications that use the console standalone library would need to setup
their own run() method that uses the optionparser if they want.

ADmad commented:
> I'm open to adding a leaner Command base class but don't have a good name for it right now.

We have `BasePlugin`,  `BaseApplication` etc. so perhaps `BaseCommand`?

saeideng commented:
> I'm open to adding a leaner Command base class but don't have a good name for it right now. Any suggestions are welcome.

for new class or for renaming `src/console/Command` to new name ?
if is new  what you want to put into it ?

markstory commented:
Renaming/changing the existing `Command` class would cause a big break for applications and I don't think it is necessary. A name like `BaseCommand` or `AbstractCommand` could work.

markstory committed:
Add BaseCommand

This class will act as a base class for the commands made without
the full stack framework.

markstory commented:
I've added a `BaseCommand` class. I'm happy to rename it if folks have an alternative name.

saeideng commented on src/Console/BaseCommand.php:
```diff
@@ -0,0 +1,271 @@
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return int|null The exit code or null for success
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
```
since we have `command` class that extended this class 
probably you can define  this method as abstract

markstory replied:
Good idea :+1:

markstory committed:
Add an abstract method to indicate what is expected of implementations.

markstory committed:
Update psalm baseline and update hint

markstory commented:
Anything else for this?

ADmad commented on src/Console/BaseCommand.php:
```diff
@@ -0,0 +1,268 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
```suggestion
     * @inheritDoc
```

markstory committed:
Update src/Console/BaseCommand.php

Co-Authored-By: ADmad <ADmad@users.noreply.github.com>
